### ISSUE TITLE
MD037/no-space-in-emphasis reported when underscore contained within emphasis

### ISSUE THREAD

**Issue by tekumara** [DavidAnson/markdownlint#324]
The following markdown snippet, reports MD037/no-space-in-emphasis

```
_~/.ssh/id_rsa_ and _foo_
```

Renders as:

_~/.ssh/id_rsa_ and _foo_

**Comment by mondeja** [comment]
It's the expected behaviour, you need to escape the `_` inside `id` and `rsa`:

`_~/.ssh/id\_rsa_ and _foo_`

[Online demo](https://dlaa.me/markdownlint/#%25m%23%20Issue%20324%0A%0A_~%2F.ssh%2Fid%5C_rsa_%20and%20_foo_%0A)

**Comment by DavidAnson** [comment]
This renders correctly because the parser makes assumptions about whitespace that allow it to ignore the embedded underscore. However, the linting rule cannot do that because it is specifically looking for mistakes in that area. Therefore, this is expected behavior for the rule. It believes the underscores match in pairs￼ and therefore sees a problem. In order to fix this, you can escape the embedded underscore with a backslash. Here's an example of your text not reporting any violations: ￼￼ https://dlaa.me/markdownlint/#%25m%23%20Issue%20324%0A%0A_~%2F.ssh%2Fid%5C_rsa_%20and%20_foo_%0A

**Comment by tekumara** [comment]
Thanks for the quick response!

It looks like the [commonmark spec](https://spec.commonmark.org/0.29/#emphasis-and-strong-emphasis) allows for underscores inside emphasis, so the parser is happy.

`_~/.ssh/id\_rsa_` makes markdownlint happy but [prettier](https://prettier.io/) removes the backslash, so I can't win 😢

**Comment by DavidAnson** [comment]
How do you feel about using asterisk for emphasis for that part? Doing so allows the underscore to be an escaped like so:￼ https://dlaa.me/markdownlint/#%25m%23%20Issue%20324%0A%0A*~%2F.ssh%2Fid_rsa*%20and%20_foo_%0A

**Comment by tekumara** [comment]
I prefer asterisk actually, but when I run prettier it converts asterisks to underscores (and causes the MD037 report) 🤕

**Comment by DavidAnson** [comment]
I've lost track, is Prettier helping us or hurting us? 😐

**Comment by tekumara** [comment]
99% of the time it's great. It will resolve a lot of issues identified by markdownlint (things like bare URLs, or trailing punctuation in headings, it can't fix automatically). For this edge case tho, markdownlint and prettier work against each other.

**Comment by DavidAnson** [comment]
I don't see a simple way for this rule to be flexible enough to detect issues and also strict enough to allow this case. Because there are two easy workarounds (escape the underscore or use asterisks), I'm going to close the issue.
